src/sardana/macroserver/macros/scan.py

```diff
@@ -1848,3 +1849,78 @@ def getTimeEstimation(self):
+            self.mntGrp = self.getObj(mntGrp, type_class=Type.MeasurementGroup)
+            channels = self.mntGrp.getChannels()
+            select_channel = ''
+
+            for channel in channels:
+                if channel['enabled'] & channel['plot_type'] == 1:
+                    select_channel = channel['label']
+                    break
+
+            # in case no channel is enabled and plotted just take the first
+            if select_channel == '':
+                select_channel = channels[0]['label']
+
+            select_motor = str(parent.motors[0])
+
+            # calculate stats for all enabled channels
+            data = parent.data
+            stats = {}
+
+            for channel in channels:
+                if channel['enabled']:
+                    channel_name = channel['label']
+                    counter_data = []
+                    motor_data = []
+
+                    for idx, rc in data.items():
+                        counter_data.append(rc[channel_name])
+                        motor_data.append(rc[select_motor])
+
+                    counter_data = numpy.array(counter_data)
+                    counter_data_grad = numpy.gradient(counter_data)
+                    motor_data = numpy.array(motor_data)
+
+                    stats[channel_name] = {'min': numpy.min(counter_data),
+                                           'max': numpy.max(counter_data),
+                                           'minpos': motor_data[numpy.argmin(counter_data)],
+                                           'maxpos': motor_data[numpy.argmax(counter_data)],
+                                           'mean': numpy.mean(counter_data),
+                                           'int': numpy.sum(counter_data),
+                                           'cen': numpy.sum(counter_data*motor_data)/numpy.sum(counter_data),
+                                           'edge': numpy.sum(counter_data_grad*motor_data)/numpy.sum(counter_data_grad)}
+
+            self.info('Statistics on channel:  %s' % select_channel)
+            self.info('Statistics for movable: %s' % select_motor)
+
+            # print statistics
+            self.info('Min:      %g' % stats[select_channel]['min'])
+            self.info('Max:      %g' % stats[select_channel]['max'])
+            self.info('Min at:   %g' % stats[select_channel]['minpos'])
+            self.info('Max at:   %g' % stats[select_channel]['maxpos'])
+            self.info('Mean:     %g' % stats[select_channel]['mean'])
+            self.info('Integral: %g' % stats[select_channel]['int'])
+            self.info('CEN:      %g' % stats[select_channel]['cen'])
+            self.info('EDGE:     %g' % stats[select_channel]['edge'])
+            # set CEN and PEAK as env variables
+            # set the motor only in case it is hard to access it from another
+            # macro like pic or cen
+            self.setEnv('ScanStats', {'stats': stats,
```

Comment:
I think it is better to set this environment variable on the door level. This way if scans are executed on two different doors they will not override the same variable.

Reply:
as I learned from #888 I could do a `self.setEnv(self.getDoorName() + '.ScanStats', {....` here.
I will apply it to the PR